test/test_time_stepping.jl

```diff
@@ -46,9 +53,22 @@ function incompressible_in_time(arch, ft, Nt)
     # Just add a temperature perturbation so we get some velocity field.
     @. model.tracers.T.data[8:24, 8:24, 8:24] += 0.01
 
-    time_step!(model, Nt, 1)
+    time_step!(model, Nt, 0.05)
+
+    velocity_div!(grid, u, v, w, div_u)
 
-    Oceananigans.velocity_div!(grid, u, v, w, div_u)
+    min_div = minimum(div_u)
+    max_div = minimum(div_u)
+    sum_div = sum(div_u)
+    abs_sum_div = sum(abs.(div_u))
+    @info "Velocity divergence after $Nt time steps ($arch, $ft): min=$min_div, max=$max_div, sum=$sum_div, abs_sum=$abs_sum_div"
 
-    all(div_u.data .< eps(ft))
+    # We are comparing with 0 so we use absolute tolerances. They are a bit larger than eps(Float64) and eps(Float32)
+    # because we are summing over the absolute value of many machine epsilons. A better atol value may be
+    # Nx*Ny*Nz*eps(ft) but it's much higher than the observed abs_sum_div.
+    if ft == Float64
+        return isapprox(abs_sum_div, 0; atol=2e-15)

```

Comment:
Is the error constant in the number of time-steps taken?

Reply:
No... It's much lower than before but something is still accumulating, just more slowly. I suspect we also have to calculate w from the continuity equation which introduces a vertical integral in the kernel that updates u, v, w but it may be needed to ensure that the velocity is divergence-free to machine precision for all time steps.

The `atol` values I used were the errors [sum of the absolute value of div(u)] at 100 time steps.

Reply:
Does the error accumulate linearly? If so, it could make sense to define the tolerance `atol = C / Nt`.

Reply:
Kind of but not exactly. That's a good idea. I'll run the test for multiple values of `Nt` and see if I can determine `C`.